lib/ansible/modules/files/read_csv.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: read_csv
+version_added: '2.8'
+short_description: Read a CSV file
+description:
+- Read a CSV file and return a list or a dictionary, containing one dictionary per row.
+author:
+- Dag Wieers (@dagwieers)
+options:
+  path:
+    description:
+    - The CSV filename to read data from.
+    type: str
+    required: yes
+    aliases: [ filename ]
+  key:
+    description:
+    - The column name used as a key for the resulting dictionary.
+    - If C(key) is unset, the module returns a list of dictionaries,
+      where each dictionary is a row in the CSV file.
+    type: str
+  dialect:
+    description:
+    - The CSV dialect to use when parsing the CSV file.
+    - Possible values include C(excel), C(excel-tab) or C(unix).
+    type: str
+    default: excel
+  fieldnames:
+    description:
+    - A list of field names for every column.
+    - This is needed if the CSV does not have a header.
+    type: list
+  unique:
+    description:
+    - Whether the C(key) used is expected to be unique.
+    type: bool
+    default: yes
+todo:
+- Make it possible to add a custom dialect
```

Comment:
That's a possibility, but I was worried that people might be getting confused. Since we don't allow all possibilities. I don't want to make the module more complex either. (Adding a custom dialect through options is a valid feature though).

Reply:
I decided not to do this, advanced.curious users can click on the "Edit on GitHub" button or look at the code to understand how the module works. The csv library documentation is not an end-user document and the relevant information for users is pretty sparse and hidden within.